### ISSUE TITLE
Make compatible with JDK 1.7 builds

### ISSUE THREAD

**Issue by Flowdalic** [unbroken-dome/gradle-testsets-plugin#1]
Is there a reason setting sourceCompatiblity to 1.8? I tried to evaluate using gradle-testsets-plugin for my project, as soon as it's configured gradle fails with

```
org/unbrokendome/gradle/plugins/testsets/TestSetsPlugin : Unsupported major.minor version 52.0
```

**Comment by unbroken-dome** [comment]
Hi,
There are some Java 8 lambda constructs in the code, which would make the build fail if called from anything lower than a JDK 8. The sourceCompatibility / targetCompatibility properties are just a fail-fast way to announce that.

The reason for this is mostly that I wrote the plugin originally in Groovy, and then ported it to Java after some issues with the sketchy Groovy support in Eclipse.

It's only 3 places however, and I can certainly rephrase those using anonymous classes instead of lambdas to make it compatible with Java 1.7.

**Comment by Flowdalic** [comment]
That would be very appreciated. I thought maybe there is a way to run gradle with sourceCompatiblity 1.8 for the buildscript only and use 1.7 (=what my project currently uses) for the artifacts. But this doesn't seem to be possible.

**Comment by Flowdalic** [comment]
Thanks for fixing this so fast. :) Would you mind pinging me once 1.0.2 is released?

**Comment by unbroken-dome** [comment]
It's already available in JCenter. The official Gradle repo always takes a while to pick up the new versions, but I can't really influence that :-)

**Comment by dbalakirev** [comment]
Hi, is it possible that this problem is present again from versions > 1.0.2? I tried with 1.1.0 and 1.2.0 and got the same error. Now this is only a problem because the 'Prerequisites' mention java 7. Thanks for your response (and for the plugin)!

**Comment by davidhatten** [comment]
I know this is closed but I've found this error again on v1.2.0. I am using Java8 for my project so I'm not sure where the issue is coming from
